app/main.py

```diff
@@ -1,13 +1,17 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {"line": error["line_number"], "column": error["column_number"],
+            "message": error["text"], "name": error["code"],
+            "source": "flake8", }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {"errors": [format_linter_error(error=error) for error in errors],
+            "path": file_path,
+            "status": "passed" if not errors else "failed", }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        format_single_linter_file(key, value)
+        for key, value in linter_report.items()
```

Comment:
Instead of using generic variable names like `key` and `value`, consider choosing more descriptive names that clearly indicate their purpose. For example, you could use `file_path` and `errors`. These names make your code easier to read.